---
description:
  'Enforces users to return types that conform to Node interface as nullable. These types can
  usually be deleted from Relay cache and schema should reflect that.'
---

# `nullable-records`

‚úÖ The `"extends": "plugin:@graphql-eslint/schema-recommended"` property in a configuration file
enables this rule.

üîß The `--fix` option on the
[command line](https://eslint.org/docs/user-guide/command-line-interface#--fix) can automatically
fix some of the problems reported by this rule.

üí° This rule provides
[suggestions](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

- Category: `Schema`
- Rule name: `@graphql-eslint/nullable-records`
- Requires GraphQL Schema: `false`
  [‚ÑπÔ∏è](/docs/getting-started#extended-linting-rules-with-graphql-schema)
- Requires GraphQL Operations: `false`
  [‚ÑπÔ∏è](/docs/getting-started#extended-linting-rules-with-siblings-operations)

{frontMatter.description}

## Usage Examples

### Incorrect

```graphql
# eslint @graphql-eslint/nullable-records: 'error'

type User {
  id: ID!
  name: String!
}

type Query {
  me: User!
}
```

### Correct

```graphql
# eslint @graphql-eslint/nullable-records: 'error'

type User {
  id: ID!
  name: String
}

type Query {
  me: User
}
```

## Config Schema

###

## Resources

- [Rule source](https://github.com/B2o5T/graphql-eslint/tree/master/packages/plugin/src/rules/nullable-records.ts)
- [Test source](https://github.com/B2o5T/graphql-eslint/tree/master/packages/plugin/__tests__/nullable-records.spec.ts)
